### PR TITLE
Clang-format our Game/Input code for GSoC

### PR DESCRIPTION
**Pull request by garbear**
## Description
Two game-related projects have been accepted to GSoC 2020: Multiplayer and Saved Game Manager. These will build on existing game and input code. Let's rip the bandaid off now, format all the game/input files they'll touch, and avoid some headaches for the students later.

I've chosen four folders to format:

* cores/RetroPlayer/
* games/
* input/
* peripherals/

These are the four folders assigned to me in [CODEOWNERS](https://github.com/xbmc/xbmc/blob/master/.github/CODEOWNERS).

This PR is 8 commits. The first 4 are the automatic formatting done by clang-format. The next 4 are fixes to the doxy indentation done by hand. Apparently clang-format doesn't de-indent doxy past the first line when de-indenting namespace declarations.

Formatted via:

```
clang-format -style=file -i xbmc/cores/RetroPlayer/*.[ch]*
clang-format -style=file -i xbmc/cores/RetroPlayer/*/*.[ch]*
clang-format -style=file -i xbmc/cores/RetroPlayer/*/*/*.[ch]*
clang-format -style=file -i xbmc/cores/RetroPlayer/*/*/*/*.[ch]*
clang-format -style=file -i xbmc/games/*.[ch]*
clang-format -style=file -i xbmc/games/*/*.[ch]*
clang-format -style=file -i xbmc/games/*/*/*.[ch]*
clang-format -style=file -i xbmc/input/*.[ch]*
clang-format -style=file -i xbmc/input/*/*.[ch]*
clang-format -style=file -i xbmc/input/*/*/*.[ch]*
clang-format -style=file -i xbmc/peripherals/*.[ch]*
clang-format -style=file -i xbmc/peripherals/*/*.[ch]*
clang-format -style=file -i xbmc/peripherals/*/*/*.[ch]*
```

## Motivation and Context
For the two game-related GSoC 2020 projects.

## How Has This Been Tested?
Not runtime tested. The last four commits were done by hand, so my eyes have visited all 266 headers that were formatted.

Compiled this change against an 18.6 branch, and observed a lot of new warnings added.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/games/addons/GameClientSubsystem.h [hunk 1]
```diff
@@ -17,60 +17,62 @@ namespace KODI
...
-     *
-     * \param subsystems The subsystems created by CreateSubsystems()
-     */
-    static void DestroySubsystems(GameClientSubsystems &subsystems);
+  /*!
+   * \brief Deallocate subsystems
+   *
+   * \param subsystems The subsystems created by CreateSubsystems()
+   */
+static void DestroySubsystems(GameClientSubsystems& subsystems);
```
Comment (Rechi) [inline xbmc/games/addons/GameClientSubsystem.h:63]: > Use spaces as tab policy with an indentation size of 2.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation

File: xbmc/input/InputCodingTableKorean.cpp [hunk 2]
```diff
@@ -147,48 +156,50 @@ std::wstring CInputCodingTableKorean::InputToKorean(const std::wstring& input)
...
       {
-        if (choseong == -1) // New consonant is choseong. Also it could be first part of double consonant.
+        if (choseong ==
+            -1) // New consonant is choseong. Also it could be first part of double consonant.
         {
-          if (jongseong != -1) // If choseong is already completed, new consonant is another choseong.
-                               // So previous character has only jongseong.
+          if (jongseong !=
+              -1) // If choseong is already completed, new consonant is another choseong.
+              // So previous character has only jongseong.
```
Comment (Rechi) [inline xbmc/input/InputCodingTableKorean.cpp:164]: > Use spaces as tab policy with an indentation size of 2.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Comment by ccope** [comment]
You might want to add these commits to a `.git-blame-ignore-revs` file after they are merged.

Example usage:
https://github.community/t5/How-to-use-Git-and-GitHub/Support-ignore-revs-file-in-Github-s-blame-view/td-p/57020

**Comment by VelocityRa** [comment]
If you change [this](https://github.com/xbmc/xbmc/blob/master/.clang-format#L30) line to `CommentPragmas:  '^\\.+'`, the doxygen comments will work fine.
No need for manual fixups.

**Comment by garbear** [comment]
@VelocityRa I changed that line to what you posted and it had no effect. Doxygen comments still required fixing by hand.

**Comment by VelocityRa** [comment]
@garbear Did you run it on the code before it was clang-formatted already? I tested it and it worked.

**Comment by garbear** [comment]
@VelocityRa I [applied]() your fix and ran:

```
clang-format -style=file -i xbmc/cores/RetroPlayer/*.[ch]*
clang-format -style=file -i xbmc/cores/RetroPlayer/*/*.[ch]*
clang-format -style=file -i xbmc/cores/RetroPlayer/*/*/*.[ch]*
clang-format -style=file -i xbmc/cores/RetroPlayer/*/*/*/*.[ch]*
```

Outcome:

```
diff --git a/xbmc/cores/RetroPlayer/RetroPlayerUtils.h b/xbmc/cores/RetroPlayer/RetroPlayerUtils.h
index f8fae72bd0..6df22eee6c 100644
--- a/xbmc/cores/RetroPlayer/RetroPlayerUtils.h
+++ b/xbmc/cores/RetroPlayer/RetroPlayerUtils.h
@@ -16,10 +16,10 @@ namespace KODI
 {
 namespace RETRO
 {
-  class CRetroPlayerUtils
-  {
-  public:
-    /*!
+class CRetroPlayerUtils
+{
+public:
+  /*!
      * \brief Convert a stretch mode enum to a short string identifier
      *
      * \param stretchMode The stretch mode
Stage this hunk [y,n,q,a,d,j,J,g,/,e,?]? 
```

You can see that doxygen comments are still not correctly de-indented.

**Comment by VelocityRa** [comment]
Hm yeah I can't recreate it now either.
It _is_ fixed if you change `ReflowComments` to `true` though (other change not needed).
But I imagine that breaks other things.

**Review by VelocityRa — APPROVED** [review]

**Comment by garbear** [comment]
Should we change `ReflowComments` to `true`? It'd be nice to not have to manually fix clang-format results.

**Comment by VelocityRa** [comment]
I would say try it iteratively on more and more directories, and see how much stuff breaks. If it's not too much and these occurrences can be fixed (perhaps by tweaking the options more), then yeah I don't see why not.

**Comment by garbear** [comment]
@VelocityRa I enabled `ReflowComments` and like the results. It entirely eliminated the manual effort, and comments seem to be better off: https://github.com/garbear/xbmc/commit/format-gsoc-test

**Comment by VelocityRa** [comment]
Nice, feel free to increase `ColumnLimit` too if you want - I see it shorted lots of comments. 100 is kinda too low anyway imo for today's screens.
If you do that, you'll likely need to reset to pre-formatted code beforehand.

**Comment by garbear** [comment]
I think that accessibility is the best line length argument. 100 characters is the typical reference I come across for vision-impaired coders, so I wouldn't change what we currently have.

**Comment by garbear** [comment]
@phunkyfish are you ok with the clang-format commit?

**Comment by phunkyfish** [comment]
I like the `ReflowComments` addition. I never knew that was an option.

I’ll will review this today.

**Comment by garbear** [comment]
I spent a lot of time experimenting with `CommentPragmas` and failed to figure it out. Even ended up in Clang source. It looks like the doxy reflow errors are rare. I pushed a fix for one error I found, and it seems clang-format doesn't undo these fixes when it is run again.

**Comment by phunkyfish** [comment]
One error with doxy in the entire codebase or just for game?

**Comment by garbear** [comment]
just for the files affected by this PR (see description), but I might have missed other errors

**Comment by phunkyfish** [comment]
So how do we tell what errors it could introduce in the rest of the codebase?

Because any future clang format will incorporate this. Is that risky?

@AlwinEsch you work a lot with Doxy. Should this be ok?

**Comment by AlwinEsch** [comment]
About doxygen, looks mostly Ok, think them need to be a bit detailed in a other request. 
For the binary addons I use mostly the groups (to have parts in right places), but to handle in whole Kodi it becomes to complex.

What I normally like is to have a line break after „@brief“ text, by „@param“ to use „[in],[out],[in/out]“. And to use a „@“ instead of „\\“, but that’s mostly for me (no requested), as a „\\“ also for other things and makes my brain aggressive for some reason 😏 (that’s no joke, no idea why 😄)

**Comment by garbear** [comment]
@phunkyfish It will only introduce errors if there is no line break, like Alwin said, and adjacent lines are very long. I think we can live with having to update these few scenarios by hand, and it's much better than having to adjust every doxygen comment like I had to before.

**Comment by phunkyfish** [comment]
Cool. Good point.

**Comment by garbear** [comment]
@phunkyfish when you check this out tomorrow, can you say if it's ready to go? GSoC is starting on June 1.

**Review by phunkyfish — APPROVED** [review]
Looks good.

**Comment by AlwinEsch** [comment]
About the `ReflowComments` is a bit bad about doxygen code styles, seen by https://github.com/xbmc/xbmc/pull/17438 that it breaks his lines.

There are few parts where no line breaks allowed for e.g. a table or there the `@defgroup`:
```cpp
  /// @defgroup cpp_kodi_addon_inputstream_Defs_Interface_InputstreamMasteringMetadata_Help Value Help
  /// @ingroup cpp_kodi_addon_inputstream_Defs_Interface_InputstreamMasteringMetadata
  ///
  /// <b>The following table contains values that can be set with @ref cpp_kodi_addon_inputstream_Defs_Interface_InputstreamMasteringMetadata :</b>
  /// | Name | Type | Set call | Get call
  /// |------|------|----------|----------
  /// | **Chromaticity X coordinates of the red** | `double` | @ref InputstreamMasteringMetadata::SetPrimaryR_ChromaticityX "SetPrimaryR_ChromaticityX" | @ref InputstreamMasteringMetadata::GetPrimaryR_ChromaticityX "GetPrimaryR_ChromaticityX"
  /// | **Chromaticity Y coordinates of the red** | `double` | @ref InputstreamMasteringMetadata::SetPrimaryR_ChromaticityY "SetPrimaryR_ChromaticityY" | @ref InputstreamMasteringMetadata::GetPrimaryR_ChromaticityY "GetPrimaryR_ChromaticityY"
  /// | **Chromaticity X coordinates of the green** | `double` | @ref InputstreamMasteringMetadata::SetPrimaryG_ChromaticityX "SetPrimaryG_ChromaticityX" | @ref InputstreamMasteringMetadata::GetPrimaryG_ChromaticityX "GetPrimaryG_ChromaticityX"
  /// | **Chromaticity Y coordinates of the green** | `double` | @ref InputstreamMasteringMetadata::SetPrimaryG_ChromaticityY "SetPrimaryG_ChromaticityY" | @ref InputstreamMasteringMetadata::GetPrimaryG_ChromaticityY "GetPrimaryG_ChromaticityY"
  /// | **Chromaticity X coordinates of the blue** | `double` | @ref InputstreamMasteringMetadata::SetPrimaryB_ChromaticityX "SetPrimaryB_ChromaticityX" | @ref InputstreamMasteringMetadata::GetPrimaryB_ChromaticityX "GetPrimaryB_ChromaticityX"
  /// | **Chromaticity Y coordinates of the blue** | `double` | @ref InputstreamMasteringMetadata::SetPrimaryB_ChromaticityY "SetPrimaryB_ChromaticityY" | @ref InputstreamMasteringMetadata::GetPrimaryB_ChromaticityY "GetPrimaryB_ChromaticityY"
  /// | **Chromaticity X coordinates of the white point** | `double` | @ref InputstreamMasteringMetadata::SetWhitePoint_ChromaticityX "SetWhitePoint_ChromaticityX" | @ref InputstreamMasteringMetadata::GetWhitePoint_ChromaticityX "GetWhitePoint_ChromaticityX"
  /// | **Chromaticity Y coordinates of the white point** | `double` | @ref InputstreamMasteringMetadata::SetWhitePoint_ChromaticityY "SetWhitePoint_ChromaticityY" | @ref InputstreamMasteringMetadata::GetWhitePoint_ChromaticityY "GetWhitePoint_ChromaticityY"
  /// | **Maximum number of bits of the display** | `double` | @ref InputstreamMasteringMetadata::SetLuminanceMax "SetLuminanceMax" | @ref InputstreamMasteringMetadata::GetLuminanceMax "GetLuminanceMax"
  /// | **Minimum number of bits of the display** | `double` | @ref InputstreamMasteringMetadata::SetLuminanceMin "SetLuminanceMin" | @ref InputstreamMasteringMetadata::GetLuminanceMin "GetLuminanceMin"
```

**Comment by garbear** [comment]
Does it work if you add newlines? like

```
//==============================================================================
/// @defgroup cpp_kodi_addon_inputstream_Defs_Interface_InputstreamMasteringMetadata class InputstreamMasteringMetadata
///
/// @ingroup cpp_kodi_addon_inputstream_Defs_Interface
///
/// @brief **Mastering metadata.**\n
///
/// Describes the metadata for [HDR10](https://en.wikipedia.org/wiki/High-dynamic-range_video).
///
```

This fixed some of the problems I saw. Otherwise we can revert the comment flow change.

**Comment by AlwinEsch** [comment]
Is in case of table not possible with newlines by doxygen :-(

**Comment by garbear** [comment]
If it breaks doxygen then let's revert.

**Pull request merged**